internal/component/loki/enrich/enrich.go

```diff
@@ -0,0 +1,196 @@
+// Package enrich provides the loki.enrich component.
+package enrich
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/loki/v3/pkg/logproto"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/common/loki"
+	"github.com/grafana/alloy/internal/component/discovery"
+	"github.com/grafana/alloy/internal/featuregate"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "loki.enrich",
```

Comment:
`loki.enrich` sounds like a good name. Initially I thought there should be a reference to `discovery` or `targets` in the name, but I gave up on that thought because in theory the targets don't have to come from a `discovery` component.